### PR TITLE
Uniq - Remove duplicated items from an array

### PR DESCRIPTION
**Pull request by TdfGomes**
This PR have the goal of adding an utility to remove duplicated items from an array with a functional approach.﻿

### CODE REVIEW

File: test/uniq.test.js [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+      {
+        email: 'nflook8@pinterest.com',
+        first_name: 'Nessa',
+        id: 9,
+        last_name: 'Flook'
+      }
+    ]
+    expect(uniqById(array)).toEqual(expected)
+  })
+  it('should remove duplicated array items and return a new array', () => {
```
Comment (helderburato) [inline test/uniq.test.js:108]: Just remember to add an empty space up here.

**Review by helderburato — CHANGES_REQUESTED** [review]
Great job @TdfGomes! Can you update the commits to follow the pattern of conventional commits?

File: docs/uniq.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+Simple array case
+```js
+const arr = [1,2,1,3,4,1,5]
+const newArr = uniq(arr) // [1,2,3,4,5]
+```
+Case with array of objects
+```js
+const arr = [{id:1,name:'maximus'},{id:2,name:'paul'},{id:1,name:'john'},{id:3,name:'anne'}]
+const uniqById = uniq('id')
+const newArray = uniqById(arr)//[{id:1,name:'maximus'},{id:2,name:'paul'}{id:3,name:'anne'}]
```
Comment (caiangums) [inline docs/uniq.md:27]: I noticed that the logic implemented filters the array and keeps just the last element from the list with the matching `id` in this case. Just checking but I think the result here should be something like:

```js
const newArray = [
  { id: 1 ,name: 'john' },
  { id: 2, name: 'paul' },
  { id: 3, name: 'anne' },
];
```

Comment (TdfGomes) [reply docs/uniq.md:27]: Yes you're right !
Will update the documentation.

File: test/uniq.test.js (hunk 1 above)
Comment (TdfGomes) [reply test/uniq.test.js:108]: 👍

**Pull request merged**